lib/iris/_lazy_data.py

```diff
@@ -126,6 +126,34 @@ def as_concrete_data(data):
     return data
 
 
+def lazy_masked_fill_value(data):
+    """
+    Return the fill value of a lazy masked array.
+
+    Args:
+
+    * data:
+        A dask array, NumPy `ndarray` or masked array
+
+    Returns:
+        The fill value of `data` if `data` represents a masked array, or None.
+
+    """
+    # If lazy, get the smallest slice of the data from which we can retrieve
+    # the fill_value.
+    if is_lazy_data(data):
+        inds = tuple([0] * (data.ndim-1))
+        smallest_slice = data[inds][:0]
+        data = as_concrete_data(smallest_slice)
```

Comment:
An interesting idea! I was going to say "Because then it won't be the smallest slice!", which turns out to not quite be accurate: 
```python
z = np.arange(24).reshape(2,3,4)
z[:0]
array([], shape=(0, 3, 4), dtype=int64)
```

Apparently (according to an offline conversation) doing the also causes the 0D case to not work properly. So, while I'm 👍 for reducing SLOC and code complexity, I think the extra line in this case adds some worthwhile resilience.